lib/services/systemd.c

```diff
@@ -155,15 +155,47 @@ systemd_daemon_reload(int timeout)
     return TRUE;
 }
 
+static bool
+systemd_mask_error(svc_action_t *op, const char *error)
+{
+    crm_trace("Could not issue %s for %s: %s", op->action, op->rsc, error);
+    if(strstr(error, "org.freedesktop.systemd1.InvalidName")
+       || strstr(error, "org.freedesktop.systemd1.LoadFailed")
+       || strstr(error, "org.freedesktop.systemd1.NoSuchUnit")) {
+
+        if (safe_str_eq(op->action, "stop")) {
+            crm_trace("Masking %s failure for %s: unknown services are stopped", op->action, op->rsc);
+            op->rc = PCMK_OCF_OK;
+
+        } else {
+            crm_trace("Mapping %s failure for %s: unknown services are not installed", op->action, op->rsc);
+            op->rc = PCMK_OCF_NOT_INSTALLED;
+            op->status = PCMK_LRM_OP_NOT_INSTALLED;
+        }
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
 static const char *
 systemd_loadunit_result(DBusMessage *reply, svc_action_t * op)
 {
     const char *path = NULL;
+    DBusError error;
 
-    if(pcmk_dbus_find_error("LoadUnit", (void*)&path, reply, NULL)) {
+    if(pcmk_dbus_find_error("LoadUnit", (void*)&path, reply, &error)) {
```

Comment:
I see a bug in pcmk_dbus_find_error() -- previously, we would never take this branch, because it always returns FALSE if its last argument is NULL. This is the only place that ever used NULL there, so this change conveniently works around the bug as a side effect.